snap/local/bin/source-env-file.sh

```diff
@@ -1,11 +1,18 @@
 #!/bin/bash -e
 
-SERVICE="app-service-configurable"
-ENV_FILE="$SNAP_DATA/config/res/$SERVICE.env"
+# convert cmdline to string array
+ARGV=($@)
+
+# grab binary path
+BINPATH="${ARGV[0]}"
+
+# binary name == service name/key
+SERVICE=$(basename "$BINPATH")
```

Comment:
Has this been tested?
I think `basename $BINPATH` is going to give you the name of this script, not the wanted name of the app. If so, please revert this change.

Reply:
I have reversed it. `basename $BINPATH` should equal `app-service-configurable` as expected, not `service-wrapper.sh`.